### ISSUE TITLE
tfsec deprecated, replaced by Trivy. Create a new hook for Trivy

### ISSUE THREAD

**Issue by bodgit** [antonbabenko/pre-commit-terraform#550]
### What problem are you facing?

tfsec [is deprecated](https://github.com/aquasecurity/tfsec/discussions/1994) and it's recommended to migrate to use Trivy instead.

### How could pre-commit-terraform help solve your problem?

A new hook that runs `trivy config ...` in some form would therefore be useful. Trivy seems to currently honour existing `tfsec:ignore:*` comment blocks so you can switch hooks without needing to rewrite existing comments.

**Comment by pdecat** [comment]
As trivy is not only for terraform, maybe it makes sense to use https://github.com/mxab/pre-commit-trivy

It was added to docs here: https://github.com/aquasecurity/trivy/pull/3203

**Comment by MaxymVlasov** [comment]
Nice, let's them just add to docs info about `tfsec` deprecation and info where can be found `trivy` hook (which maintenance will be not our problem 🎉 )

**Comment by MaxymVlasov** [comment]
Of course, if repo author will add a LICENCE, otherwise his hooks are not usable

**Comment by bodgit** [comment]
I'd actually rather not use that hook as it requires Docker and is pinned to the version of trivy specified in the hook configuration, which means I'm reliant on the author to keep updating the hook.

Given trivy is a single binary, I can (and do) install that by other means, so a hook that just tries to run `trivy` and assume it's somewhere in `$PATH` would be more flexible.

**Comment by pdecat** [comment]
Maybe that's something the author of https://github.com/mxab/pre-commit-trivy/ would accept to implement, especially easy to add without breaking anything IMO as the current hooks are suffixed by `-docker`.

**Comment by MaxymVlasov** [comment]
Well, then you can just copy-paste [`tfsec` hook](https://github.com/antonbabenko/pre-commit-terraform/blob/master/hooks/terraform_tfsec.sh), change `tfsec` execution command, and that mostly it from hook addition prospective

There is a full doc on how to add docs and so on - https://github.com/antonbabenko/pre-commit-terraform/blob/master/.github/CONTRIBUTING.md#add-new-hook

Also, we will need a deprecation notice to `tfsec` both in README and in hook itself

The hook notice can be just 
```bash
common::colorify "yellow" "tfsec tool was deprecated, and replaced by trivy. You can check trivy hook here:"
common::colorify "yellow" "https://github.com/antonbabenko/pre-commit-terraform/tree/master#terraform_trivy"
```
at the [end of `main` function](https://github.com/antonbabenko/pre-commit-terraform/blob/66a1469a7c7954d8469d7a6f79f4ffb8c9ae09bf/hooks/terraform_tfsec.sh#L25) (of right before `common::per_dir_hook` execution - not sure that it will shown after)

**Comment by MaxymVlasov** [comment]
> Is this waiting for someone to actually come with the PR for the mentioned changes?
Exactly. And it can be you :)

**Comment by antonbabenko** [comment]
This issue has been resolved in version 1.85.0 :tada:
